### PR TITLE
Fix up containers-storage.conf man page for devicemapper options

### PR DESCRIPTION
**Pull request by rhatdan**
Need to document the devicemapper changes.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @nalind PTAL

File: docs/containers-storage.conf.5.md [hunk 1]
```diff
@@ -56,6 +56,76 @@ The `storage.options` table supports the following options:
...
+    2: LogLevelFatal
+    3: LogLevelErr
+    4: LogLevelWarn
+    5: LogLevelNotice
+    6: LogLevelInfo
+    7: LogLevelDebug
+
+**min_free_space**=""
+
+Specifies the min free space percent in a thin pool require for new device creation to succeed. Valid values are from 0% - 99%. Value 0% disables (10% is the default)
```
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md]: require to required

File: docs/containers-storage.conf.5.md [hunk 2]
```diff
@@ -56,6 +56,76 @@ The `storage.options` table supports the following options:
...
+
+Specifies the size to use when creating the base device, which limits the size of images and containers. (10g is the default)
+
+**blocksize**=""
+
+Specifies a custom blocksize to use for the thin pool. (64k is the default)
+
+**directlvm_device**=""
+
+Specifies a custom block storage device to use for the thin pool. Required if you setup devicemapper
```
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md]: nit , missing end period.

File: docs/containers-storage.conf.5.md [hunk 3]
```diff
@@ -56,6 +56,76 @@ The `storage.options` table supports the following options:
...
+
+**directlvm_device_force**=""
+
+Tells driver to wipe device (directlvm_device) even if device already has a filesystem.  Default is False
```
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md]: nit , missing end period.

File: docs/containers-storage.conf.5.md (hunk 1 above)
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md]: nit , missing end period.

File: docs/containers-storage.conf.5.md [hunk 4]
```diff
@@ -56,6 +56,76 @@ The `storage.options` table supports the following options:
...
+
+Marks device for deferred removal.  If the device is in use when it driver attempts to remove it, driver will tell the kernel to remove it as soon as possible.  (Default is true).
+
+**use_deferred_deletion**=""
+
+Marks device for deferred deletion. If the device is in use when it driver attempts to delete it, driver continue to attempt to delete device every 30 seconds, or when it restarts.  (Default is true).
+
+**xfs_nospace_max_retries**=""
+
+Specifies the maximum number of retries XFS should attempt to complete IO when ENOSPC (no space) error is returned by underlying storage device. (Default is 0, which means to try continuously.
```
Comment (TomSweeneyRedHat) [inline docs/containers-storage.conf.5.md]: Missing end paren for Default statement.

File: docs/containers-storage.conf.5.md (hunk 1 above)
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: Fixed

File: docs/containers-storage.conf.5.md (hunk 2 above)
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: Fixed

File: docs/containers-storage.conf.5.md (hunk 3 above)
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: FIxed

File: docs/containers-storage.conf.5.md (hunk 1 above)
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: Fixed

File: docs/containers-storage.conf.5.md (hunk 4 above)
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: Fixed

File: storage.conf [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# storage.conf is the configuration file for all tools
+# that share the containers/storage libraries
 # See man 5 containers-storage.conf for more information
 # The "container storage" table contains all of the server options.
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply storage.conf]: We ship containers-storage.conf.5, but we install /etc/containers/storage.conf.

 So I changed the first line to say this file.

**Comment by TomSweeneyRedHat** [comment]
I'm not seeing any of the fixed nits in the latest?

**Comment by rhatdan** [comment]
@TomSweeneyRedHat They should be there now.

File: storage.conf [hunk 6]
```diff
@@ -93,17 +95,24 @@ override_kernel_check = "false"
...
+# mkfsarg specifies extra mkfs arguments to be used when creating the base.
 # device.
 # mkfsarg = ""
 
 # mountopt specifies extra mount options used when mounting the thin devices.
 # mountopt = ""
 
-# use_deferred_removal Marking device for deferred removal
+# use_deferred_removal marks devicemapper block device for deferred removal.
+# If the device is in use when it driver attempts to remove it, driver will
```
Comment (TomSweeneyRedHat) [inline storage.conf]: "it driver" to "its driver", also ", driver will" to ", the driver will"

File: storage.conf [hunk 7]
```diff
@@ -93,17 +95,24 @@ override_kernel_check = "false"
...
+# tell the kernel to remove it as soon as possible. Note this does not free
+# up the disk space, see deferred deletion for this.
 # use_deferred_removal = "True"
 
-# use_deferred_deletion Marking device for deferred deletion
+# use_deferred_deletion marks thinpool device for deferred deletion.
+# If the device is in use when it driver attempts to delete it, driver
```
Comment (TomSweeneyRedHat) [inline storage.conf]: it to its and added 'the' here too.

File: docs/containers-storage.conf.5.md [hunk 8]
```diff
@@ -48,14 +48,84 @@ The `storage.options` table supports the following options:
...
+
+Specifies extra mkfs arguments to be used when creating the base device.
+
+**mountopt**=""
+
+Specifies extra mount options used when mounting the thin devices.
+
+**use_deferred_removal**=""
+
+Marks devicemapper block device for deferred removal.  If the device is in use when it driver attempts to remove it, driver will tell the kernel to remove it as soon as possible.  Note this does not free up the disk space, see deferred deletion for this.  (Default is true).
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: fixed

File: docs/containers-storage.conf.5.md [hunk 9]
```diff
@@ -48,14 +48,84 @@ The `storage.options` table supports the following options:
...
+
+**use_deferred_deletion**=""
+
+Marks thinpool device for deferred deletion. If the device is in use when it driver attempts to delete it, driver continue to attempt to delete device every 30 seconds, or when it restarts.  Deferred deletion permanently deletes the device and all data in the device will be lost. (Default is true).
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/containers-storage.conf.5.md]: Fixed

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM and happy green test buttons.  Thanks for the nit extermination.

**Pull request merged**